aroquev/src/main/webapp/script.js

```diff
@@ -212,9 +214,39 @@ function getServletMessage() {
   });
 }
 
+/**
+ * Verify the input number from selection of number of comments to display
+ * Minimum number is 1 comment, and maximum is 10 comments
+ */
+function verifyNumberCommentsInput() {
+  // Default value
+  let maxComments = 5;
+
+  let maxCommentsInput = document.getElementById("maxComments").value;
+  const commentsErrorContainer = document.getElementById('js-comments-input-error-container');
+  if (maxCommentsInput.length != 0 && maxCommentsInput < 11 && maxCommentsInput > 0) {
+    maxComments = maxCommentsInput;
+    commentsErrorContainer.innerText = "";
+  }
+  else {
+    commentsErrorContainer.innerText = "Invalid input for number of comments!";
+  }
+
+  return maxComments;
+}
+
 /** Creates an <li> element containing text. */
 function createListElement(text) {
   const liElement = document.createElement('li');
   liElement.innerText = text;
   return liElement;
 }
+
+/**
+ * Deletes all comments in DS and then refreshes comments section
+ */
+function deleteAllComments() {
+  fetch('/delete-data', {method: 'POST'}).then(() => {
+    getServletComments();
+  });
```

Comment:
Here the response from the server gets sent once it has executed all of its code? Or is it possible that it returns a response while it has not yet deleted all comments?

Reply:
Yes, I think that the server should only respond when the request handler has finished executing, so all comments should be deleted in this case. Do you see something else?

Reply:
Nope, just in some random cases the page reloads and still displays some comments, but this rarely happens and only in my dev-server.

Reply:
It could have something to do with the way Datastore works. It's not guaranteed that a read immediately following a write will return consistent data. You can read more about this here https://cloud.google.com/datastore/docs/concepts/structuring_for_strong_consistency

Reply:
That link applies to the older datastore. Your project should be using "Firestore in Datastore mode", which has strong consistency.

Reply:
Mmm yep you're right, so all our queries should have strong consistency?

Reply:
That should be the case. If you read before a write is finished you'd get stale data - with or without strong consistency.